File: MaxConsecutiveOnes.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        
        int maxCount = 0;
        int count = 0;

        for(int num : nums){
            if(num == 1){
                count++;
                maxCount = Math.max(count, maxCount);
            }else{
                count = 0;
            }
        }
        return maxCount;
    }
}

File: MaxConsecutiveOnes.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let max = 0;
    let freq = 0;
    nums.forEach((num) => {
        freq = (num === 1) ? freq + 1 : 0;
        max = Math.max(max, freq);
    });
    return max;
};
